### PR TITLE
upgrade: Add an option to identify the version of product being upgraded

### PR DESCRIPTION
**Pull request by jsuchome**
Some upgrade commands might be different for different products being upgraded.
So let's read the system status to identify ongoing upgrade version
(if there is any) and allow user to provide specific option via ENV variable
(for the cases crowbarctl is run remotely).

### CODE REVIEW

File: lib/crowbar/client/config.rb [hunk 1]
```diff
@@ -93,6 +94,33 @@ def values
...
+        return "7-8" if File.exist?("/var/lib/crowbar/upgrade/7-to-8-upgrade-running")
+
+        # if upgrade has not been started, check the system version
+        os_release_file = "/etc/os-release"
+
+        if File.exist?(os_release_file)
+          os_release = Hash[
+            File.open(os_release_file).read.scan(/(\S+)\s*=\s*"([^"]+)/)
+          ]
+          puts "os_release: #{os_release}"
```
Comment (skazi0) [inline lib/crowbar/client/config.rb]: where will this `puts` go?

Comment (jsuchome) [reply lib/crowbar/client/config.rb]: sorry, debug output

**Comment by nicolasbock** [comment]
I have to agree with @rhafer that it looks a little strange. There is nothing technically wrong with it, but it looks strange because I and presumably @rhafer immediately translate all possible values to the same `-1` :wink:.

**Comment by jsuchome** [comment]
@rhafer @nicolasbock in general I do not much care and could use something different. However, we're using 6-7 / 7-8 in other places already, like in the status and running files: 
https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/controllers/application_controller.rb#L37

https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/lib/crowbar/upgrade_status.rb#L34

So I think it's at least consistent...

**Comment by rhafer** [comment]
> https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/controllers/application_controller.rb#L37
> 
> https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/lib/crowbar/upgrade_status.rb#L34
> 
> So I think it's at least consistent...

Actually both examples use `7-to-8`. So neither `7-8` nor `7to8`.  ;-)

**Comment by jsuchome** [comment]
Now you got me

**Comment by jsuchome** [comment]
@rhafer @nicolasbock @skazi0 Changed again and hopefully for the last time

**Pull request merged**